cluster-autoscaler/processors/processors.go

```diff
@@ -89,6 +89,8 @@ func DefaultProcessors(options config.AutoscalingOptions) *AutoscalingProcessors
 		}),
 		ScaleUpStatusProcessor: status.NewDefaultScaleUpStatusProcessor(),
 		ScaleDownNodeProcessor: nodes.NewPreFilteringScaleDownNodeProcessor(),
+		// TODO NewMaxNodesProcessor should be called after NewAtomicResizeFilteringProcessor
```

Comment:
I wanted to put the TODO comment here to discuss if there is any reason we use MaxNodesProcessor first, created issue for it https://github.com/kubernetes/autoscaler/issues/7342

Reply:
The problem with making MaxNodesProcessor last is that it is unaware of atomic node groups, so it could filter out a single node from an atomic node group. Seems like we cannot easily separate the concerns between these two processors without making assumptions about ordering and so they should perhaps be a single object? I think the bottom line is that whatever composite processor we build, it should be aware of the subprocessors (and unit test for the interactions between them), because these subprocessors are not independent.